src/x86/intel_edison_fab_c.c

```diff
@@ -373,8 +374,17 @@ mraa_intel_edison_aio_init_post(mraa_aio_context dev)
 mraa_result_t
 mraa_intel_edison_pwm_enable_pre(mraa_pwm_context dev, int enable) {
     // PWM 0% duty workaround: update state array
-    // if someone first ran write(0) and then enable(1).
-    if ((pwm_disabled[dev->pin] == 1) && (enable == 1)) { pwm_disabled[dev->pin] = 0; }
+    if (enable == 1) {
+        pwm_disabled[dev->pin] = 0;
+    } else {
+        pwm_disabled[dev->pin] = 1;
+        // Set duty cycle to 0 before disable pwm.
+        // Edison PWM output stuck at high if disabled during ON period
+        mraa_pwm_pulsewidth_us(dev, 0);
+        // Sleep 2 period to allow change take effect
+        usleep(dev->period >> 9);
```

Comment:
This means we are adding an additional delay, which again can break or alter the behavior of the existing code.

Are you at all sure the reason for this "stuck at 100%" is that the PWM is being disabled during the duty cycle? How did you come to that conclusion? My measurements (see comment in #585) do not confirm that as the failure rate should have been increasing with the duty increase, but it hasn't.

Finally - interesting way to calculate the two periods' time - not providing the exact 2x value AFAICS, why not simply multiply? Shifting is typically faster, but in this context it's unlikely to make any difference, plus that imprecision... I wonder what's the idea behind this.